pkg/operation/botanist/component/kubeapiserver/dependency_watchdog.go

```diff
@@ -54,54 +51,53 @@ func DependencyWatchdogEndpointConfiguration() (map[string]restarterapi.Service,
 	}, nil
 }
 
-// DependencyWatchdogProbeConfiguration returns the configuration for the dependency watchdog (probe role)
-// ensuring that its dependant pods are scaled as soon a probe fails.
-func DependencyWatchdogProbeConfiguration() ([]scalerapi.ProbeDependants, error) {
-	return []scalerapi.ProbeDependants{{
-		Name: "shoot-" + v1beta1constants.DeploymentNameKubeAPIServer,
-		Probe: &scalerapi.ProbeConfig{
-			External:      &scalerapi.ProbeDetails{KubeconfigSecretName: dependencywatchdog.ExternalProbeSecretName},
-			Internal:      &scalerapi.ProbeDetails{KubeconfigSecretName: dependencywatchdog.InternalProbeSecretName},
-			PeriodSeconds: pointer.Int32(30),
+// NewDependencyWatchdogProberConfiguration returns the configuration for the dependency watchdog (probe role)
+// ensuring that its dependant pods are scaled as soon a prober fails.
+func NewDependencyWatchdogProberConfiguration() ([]proberapi.DependentResourceInfo, error) {
+	return []proberapi.DependentResourceInfo{
+		{
+			Ref: &autoscalingv1.CrossVersionObjectReference{
+				Kind:       "Deployment",
+				Name:       v1beta1constants.DeploymentNameKubeControllerManager,
+				APIVersion: "apps/v1",
+			},
+			Optional: false,
+			ScaleUpInfo: &proberapi.ScaleInfo{
+				Level: 0,
+			},
+			ScaleDownInfo: &proberapi.ScaleInfo{
+				Level: 1,
+			},
 		},
-		DependantScales: []*scalerapi.DependantScaleDetails{
-			{
-				ScaleRef: autoscalingv1.CrossVersionObjectReference{
-					APIVersion: appsv1.SchemeGroupVersion.String(),
-					Kind:       "Deployment",
-					Name:       v1beta1constants.DeploymentNameKubeControllerManager,
-				},
-				ScaleUpDelaySeconds: pointer.Int32(120),
+		{
+			Ref: &autoscalingv1.CrossVersionObjectReference{
+				Kind:       "Deployment",
+				Name:       v1beta1constants.DeploymentNameMachineControllerManager,
+				APIVersion: "apps/v1",
```

Comment:
Can you align this with https://github.com/gardener/gardener/pull/6693/files#diff-de582b2c58a37e27297fb8032a91b8836b77be1a49d0ae3ead219e25910bc77bR91?